### PR TITLE
Bump Jenkins to 2.289.1

### PR DESCRIPTION
**Pull request by khughes-atlassian**
* Add dependency management on bom and remove versions that are managed by that
* Harden acceptance tests
* Remove dependency on trilead-ssh2 (it seems to not be needed)

### CODE REVIEW

File: acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/pageobjects/BitbucketScmConfig.java [hunk 1]
```diff
@@ -23,36 +23,47 @@ public BitbucketScmConfig(Job job, String path) {
     }
 
     public BitbucketScmConfig credentialsId(String credentialsId) {
+        scrollIntoView(this.credentialsId);
```
Comment (khughes-atlassian) [inline acceptance-tests/src/main/java/it/com/atlassian/bitbucket/jenkins/internal/pageobjects/BitbucketScmConfig.java:26]: Most of the failing acceptance tests were due to the fact that the element was obscured by something in the UI

File: pom.xml [hunk 2]
```diff
@@ -37,6 +37,18 @@
         </license>
     </licenses>
 
+    <dependencyManagement>
+        <dependencies>
+            <dependency>
+                <groupId>io.jenkins.tools.bom</groupId>
+                <artifactId>bom-2.289.x</artifactId>
```
Comment (khughes-atlassian) [inline pom.xml:44]: Hopefully this means that next time we bump the minimum Jenkins version we don't have to also find out all the associated dependencies that need to be bumped.

File: pom.xml [hunk 3]
```diff
@@ -95,36 +107,31 @@
             <artifactId>httpcore</artifactId>
             <version>4.4.14</version>
         </dependency>
-        <dependency>
-            <groupId>org.jenkins-ci</groupId>
-            <artifactId>trilead-ssh2</artifactId>
```
Comment (khughes-atlassian) [inline pom.xml:100]: This didn't seem to be used anywhere. Also, there's no updated version and one of its dependencies clashes with other dependencies being brought in. I could probably just exclude that one, but then I thought removing it entirely is probably cleaner

File: pom.xml [hunk 4]
```diff
@@ -95,36 +107,31 @@
...
-            <version>build-217-jenkins-27</version>
-        </dependency>
         <dependency>
             <groupId>org.jenkins-ci.modules</groupId>
             <artifactId>instance-identity</artifactId>
-            <version>2.2</version>
+            <!--<version>managed by bom</version>-->
```
Comment (khughes-atlassian) [inline pom.xml:113]: I thought that something obvious like this to show where the version is coming from would reduce confusion in the future, but happy to remove these comments entirely too.

File: pom.xml [hunk 5]
```diff
@@ -17,8 +17,8 @@
         <disableTestInjection>true</disableTestInjection>
         <hamcrest.version>2.2</hamcrest.version>
         <java.level>8</java.level>
-        <jackson.version>2.11.2</jackson.version>
-        <jenkins.version>2.249.1</jenkins.version>
+        <jackson.version>2.13.0</jackson.version>
+        <jenkins.version>2.289.1</jenkins.version>
```
Comment (dkjellin) [inline pom.xml:21]: As mentioned duing our chat, I think it would be good to have a comment here so we update the `io.jenkins.tools.bom` at the same time.

**Review by dkjellin — APPROVED** [review]
Looks, you also need to update the README with the minimum version update. 
Since all tests are green I assume we no longer need the `trilead-ssh2` but I defer to 
@mhenschke-atl  for a authoritative decision on it.

File: readme.md [hunk 6]
```diff
@@ -26,7 +26,7 @@ The plugin streamlines the entire configuration process and removes the need for
 
 ## Requirements
 
-- Jenkins 2.249.1+
+- Jenkins 2.289.1+
```
Comment (dkjellin) [inline readme.md:29]: You also need a changelog entry, sorry.

File: src/test/java/it/com/atlassian/bitbucket/jenkins/internal/fixture/JenkinsProjectHandler.java [hunk 7]
```diff
@@ -121,10 +127,11 @@ public void runWorkflowJobForBranch(WorkflowMultiBranchProject mbp, String branc
         Future<WorkflowRun> startCondition = item.scheduleBuild2(0).getStartCondition();
         WorkflowRun workflowRun = startCondition.get(1, TimeUnit.MINUTES);
 
-        while (!workflowRun.equals(item.getLastSuccessfulBuild())) {
-            System.out.println("Waiting for workflow run to finish");
+        while (!workflowRun.equals(item.getLastCompletedBuild())) {
```
Comment (khughes-atlassian) [inline src/test/java/it/com/atlassian/bitbucket/jenkins/internal/fixture/JenkinsProjectHandler.java:130]: When something goes wrong in the run, the test doesn't fail until it times out. By using `getLastCompletedBuild` and checking that it is successful afterwards we're able to fail a bit faster

**Review by dkjellin — APPROVED** [review]

**Review by dkjellin — APPROVED** [review]

**Pull request merged**